### PR TITLE
Update to 0.13, update deps, and fix alert updates.

### PR DESCRIPTION
**Pull request by pcn**
This results in a build that almost without changes. Running tests found issues which are fixed.

There are 2 example files now, one for 0.12 syntax, and one for 0.11 syntax.

This was built with the current 0.13.0 version of the terraform library to not be too out of date.

### CODE REVIEW

**Comment by pcn** [comment]
This should pass all current tests. This adds an environment variable so that request and responses will only display when `TF_AO_DEBUG` is set. I have no attachment to that name, it's probably best if it changes.

```
spacey@bagpig:~/go/src/github.com/appoptics/terraform-provider-appoptics (upgrade-to-0.12)$ TF_ACC=1 APPOPTICS_TOKEN='<token>' go test -v ./... 
?   	github.com/appoptics/terraform-provider-appoptics	[no test files]
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestProvider_impl
--- PASS: TestProvider_impl (0.00s)
=== RUN   TestAccAppOpticsAlertMinimal
--- PASS: TestAccAppOpticsAlertMinimal (1.86s)
=== RUN   TestAccAppOpticsAlertBasic
--- PASS: TestAccAppOpticsAlertBasic (1.37s)
=== RUN   TestAccAppOpticsAlertFull
--- PASS: TestAccAppOpticsAlertFull (3.50s)
=== RUN   TestAccAppOpticsAlertUpdated
--- PASS: TestAccAppOpticsAlertUpdated (10.99s)
=== RUN   TestAccAppOpticsAlertRename
--- PASS: TestAccAppOpticsAlertRename (11.04s)
=== RUN   TestAccAppOpticsAlertFullUpdate
--- PASS: TestAccAppOpticsAlertFullUpdate (3.55s)
=== RUN   TestAccAppOpticsMetrics
--- PASS: TestAccAppOpticsMetrics (31.81s)
=== RUN   TestAccAppOpticsServiceBasic
--- PASS: TestAccAppOpticsServiceBasic (2.44s)
=== RUN   TestAccAppOpticsServiceUpdated
--- PASS: TestAccAppOpticsServiceUpdated (12.03s)
=== RUN   TestAccAppOpticsSpaceChartBasic
--- FAIL: TestAccAppOpticsSpaceChartBasic (0.01s)
    testing.go:572: Step 0 error: config is invalid: 2 problems:
        
        - Invalid resource type: The provider provider.appoptics does not support resource type "appoptics_space_chart".
        - Invalid resource type: The provider provider.appoptics does not support resource type "appoptics_space".
=== RUN   TestAccAppOpticsSpaceChart_Full
--- FAIL: TestAccAppOpticsSpaceChart_Full (0.01s)
    testing.go:572: Step 0 error: config is invalid: 3 problems:
        
        - Invalid resource type: The provider provider.appoptics does not support resource type "appoptics_space_chart".
        - Invalid resource type: The provider provider.appoptics does not support resource type "appoptics_space".
        - Invalid resource type: The provider provider.appoptics does not support resource type "appoptics_space".
=== RUN   TestAccAppOpticsSpaceChart_Updated
--- FAIL: TestAccAppOpticsSpaceChart_Updated (0.01s)
    testing.go:572: Step 0 error: config is invalid: 2 problems:
        
        - Invalid resource type: The provider provider.appoptics does not support resource type "appoptics_space_chart".
        - Invalid resource type: The provider provider.appoptics does not support resource type "appoptics_space".
=== RUN   TestAccAppOpticsSpaceBasic
--- FAIL: TestAccAppOpticsSpaceBasic (0.01s)
    testing.go:572: Step 0 error: config is invalid: Invalid resource type: The provider provider.appoptics does not support resource type "appoptics_space".
FAIL
FAIL	github.com/appoptics/terraform-provider-appoptics/appoptics	78.627s
FAIL
```

File: appoptics/resource_appoptics_metric.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ func resourceAppOpticsMetric() *schema.Resource {
 			},
 			"type": {
 				Type:     schema.TypeString,
-				Optional: true,
+				Required: true,
```
Comment (akahn) [inline appoptics/resource_appoptics_metric.go:29]: This change seems unrelated to the goal of this PR. Am I mistaken?

Comment (pcn) [reply appoptics/resource_appoptics_metric.go:29]: Without this being specified in the resource, the API will send back changes that break tests/plans.

Comment (akahn) [reply appoptics/resource_appoptics_metric.go:29]: Huh. How did it work before, then? Did the API change?

Comment (akahn) [reply appoptics/resource_appoptics_metric.go:29]: Is there any chance this could break existing users' metric definitions, since they are not required to specify their metric type. In other words, this seems like a breaking change.

Comment (pcn) [reply appoptics/resource_appoptics_metric.go:29]: Put it in the changelog, pending writing up a better README with version info.
